SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -7820,182 +7820,14 @@ void CEulerSolver::Evaluate_ObjFunc(CConfig *config) {
-
-      numerics->GetInviscidProjFlux(&Density_b, Velocity_b, &Pressure_b, &Enthalpy_b, NormalArea, Residual);
-
-      /*--- Grid velocity correction to the energy term ---*/
-      if (grid_movement) {
-        GridVel = geometry->node[iPoint]->GetGridVel();
-        ProjGridVel = 0.0;
-        for (iDim = 0; iDim < nDim; iDim++)
-          ProjGridVel += GridVel[iDim]*UnitNormal[iDim];
-        Residual[nVar-1] += Pressure_b*ProjGridVel*Area;
-      }
-
-      /*--- Add the Reynolds stress tensor contribution ---*/
-
-      if (tkeNeeded) {
-        for (iDim = 0; iDim < nDim; iDim++)
-          Residual[iDim+1] += (2.0/3.0)*Density_b*turb_ke*NormalArea[iDim];
-      }
-      
-      /*--- Add value to the residual ---*/
-      
-      LinSysRes.AddBlock(iPoint, Residual);
-      
-      /*--- Form Jacobians for implicit computations ---*/
-      
-      if (implicit) {
-        
-        /*--- Initialize Jacobian ---*/
-        
-        for (iVar = 0; iVar < nVar; iVar++) {
-          for (jVar = 0; jVar < nVar; jVar++)
-            Jacobian_i[iVar][jVar] = 0.0;
-        }
-        
-        /*--- Compute DubDu ---*/
-
-        for (iVar = 0; iVar < nVar; iVar++) {
-          for (jVar = 0; jVar < nVar; jVar++)
-            DubDu[iVar][jVar]= 0.0;
-          DubDu[iVar][iVar]= 1.0;
-        }
-
-        for (iDim = 0; iDim < nDim; iDim++)
-          for (jDim = 0; jDim<nDim; jDim++)
-            DubDu[iDim+1][jDim+1] -= UnitNormal[iDim]*UnitNormal[jDim];
-        DubDu[nVar-1][0] += 0.5*ProjVelocity_i*ProjVelocity_i;
-        for (iDim = 0; iDim < nDim; iDim++) {
-          DubDu[nVar-1][iDim+1] -= ProjVelocity_i*UnitNormal[iDim];
-        }
-
-        /*--- Compute flux Jacobian in state b ---*/
-
-        numerics->GetInviscidProjJac(Velocity_b, &Enthalpy_b, &Chi_b, &Kappa_b, NormalArea, 1, Jacobian_b);
-
-        // Check for grid movement, should be already considered since Jacobian b is computed from u_b
-        // if (grid_movement) {
-        // Jacobian_b[nVar-1][0] += 0.5*ProjGridVel*ProjGridVel;
-        // for (iDim = 0; iDim < nDim; iDim++)
-        // Jacobian_b[nVar-1][iDim+1] -= ProjGridVel * UnitNormal[iDim];
-        // }
-
-        /*--- Compute numerical flux Jacobian at node i ---*/
-
-        for (iVar = 0; iVar < nVar; iVar++)
-          for (jVar = 0; jVar < nVar; jVar++)
-            for (kVar = 0; kVar < nVar; kVar++)
-              Jacobian_i[iVar][jVar] += Jacobian_b[iVar][kVar] * DubDu[kVar][jVar];
-
-        /*--- Add the Jacobian to the sparse matrix ---*/
-
-        Jacobian.AddBlock(iPoint, iPoint, Jacobian_i);
-
-      }
-    }
-  }
-  
-  delete [] Normal;
-  delete [] NormalArea;
-  delete [] Velocity_b;
-  delete [] Velocity_i;
-  for (iVar = 0; iVar < nVar; iVar++) {
-    delete [] Jacobian_b[iVar];
-    delete [] DubDu[iVar];
-  }
-  delete [] Jacobian_b;
-  delete [] DubDu;
+  BC_Sym_Plane(geometry, solver_container, conv_numerics, visc_numerics, config, val_marker);
```

Comment:
So, I understand that this is the only change needing some additional attention to make sure that the non-ideal cases are handled for inviscid flows or slip walls in viscous problems, correct? @AlbertoGuardone: if you or anyone in your group has a moment, can you please take a look?

Reply:
Ciao @economon, ciao @TobiKattmann, we looked into it and it sounds good. 
Implementation is correct: it uses a ghost node with an inverted velocity which does not changes the thermodynamic state. The Jacobian is OK as well. Green light from a NICFD perspective.
Thanks!

Reply:
Thanks Alberto